Mathlib/Analysis/InnerProductSpace/Simultaneous.lean

```diff
@@ -0,0 +1,123 @@
+/-
+Copyright (c) 2024 Jon Bannon. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Jon Bannon, Jack Cheverton, Samyak Dhar Tuladhar
+-/
+
+import Mathlib.Analysis.InnerProductSpace.Spectrum
+import Mathlib.Analysis.InnerProductSpace.Projection
+import Mathlib.Order.CompleteLattice
+
+/-! # Simultaneous eigenspaces of a commuting pair of symmetric operators
+
+This file collects various decomposition results for simultaneous eigenspaces of a commuting pair
+of symmetric operators on a finite-dimensional Hilbert space.
+
+# Main Result
+
+* `DirectSum.IsInternal_of_simultaneous_eigenspaces_of_commuting_symmetric_pair` establishes that
```

Comment:
I think this name is too long otherwise. Note also, we're beginning to use dot notation even if there are no arguments with that type as input, because we can still use it with the *expected type* dot notation.
```suggestion
* `DirectSum.IsInternal.eigenspaces_of_commuting_symmetric_pair` establishes that
```